### PR TITLE
Update qsim demo + title

### PR DESCRIPTION
**Pull request by thisac**
I realised that I somewhat incorrectly stated that the distribution that the circuit followed is a Porter-Thomas distribution while, in fact, it only converges to such a distribution for many qubits. Clarified this section a bit.

Also updated the title to "Beyond classical computing with qsim", including in metadata.

### CODE REVIEW

File: demonstrations/qsim_supremacy.py [hunk 1]
```diff
@@ -341,17 +341,18 @@ def circuit(seed=42, return_probs=False):
...
 #
 # .. math::
 #
 #    Pr(p) = (N - 1)(1- p)^{N-2},
 #
-# where :math:`N = 2^n` is the number of possible bitstrings.
-# From this we can then calculate the expectation value
+# where :math:`N = 2^n` is the number of possible bitstrings [#Boixo2018]_.
+# This distribution is approximated well by the Porter-Thomas distribution,
+# given by :math:`Pr(p) = Ne^{-Np}`; a characteristic of chaotic quantum
```
Comment (glassnotes) [inline demonstrations/qsim_supremacy.py:354]:
```suggestion
# given by :math:`Pr(p) = Ne^{-Np}`, a characteristic of chaotic quantum
```

File: demonstrations/qsim_supremacy.py [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
 .. _quantum_neural_net:
 
-Quantum supremacy using qsim
-============================
+Beyond classical computing with qsim
+====================================
 
 .. meta::
-    :property="og:description": Discover quantum supremacy using Google's qsim simulator, and recreate their benchmarks and circuits.
+    :property="og:description": Use Google's qsim simulator to go beyond classical computing, and recreate their benchmarks and circuits.
```
Comment (co9olguy) [inline demonstrations/qsim_supremacy.py:9]:
```suggestion
    :property="og:description": Use Google's qsim simulator to explore the barriers between quantum and classical computing, and recreate their benchmarks and circuits.
```
still think it sounds weird to use a classical computer program to go beyond classical computing :stuck_out_tongue_closed_eyes:

File: demos_research.rst [hunk 3]
```diff
@@ -124,7 +124,7 @@ plateaus in quantum neural networks.
 :html:`<div class="gallery-grid row">`
 
 .. customgalleryitem::
-    :tooltip: Quantum supremacy with qsim.
+    :tooltip: Beyond classical computing with qsim.
     :figure: demonstrations/qsim_supremacy/sycamore.png
     :description: :doc:`demos/qsim_supremacy`
```
Comment (glassnotes) [inline demos_research.rst:129]: Title of the files/directories should also be changed to match

File: demonstrations/qsim_supremacy.py (hunk 1 above)
Comment (co9olguy) [inline demonstrations/qsim_supremacy.py:354]:
```suggestion
# given by :math:`Pr(p) = Ne^{-Np}`, a characteristic property of chaotic quantum
```

**Review by co9olguy — APPROVED** [review]

**Pull request merged**